include/input_handler.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+#include <window.h>
+#include <functional>
+
+namespace arch{
+    class InputHandler{
+    public:
+        InputHandler(const InputHandler&) = delete;
+        InputHandler(const InputHandler&&) = delete;
+        static InputHandler& get();
+        void set_current_window(GLFWwindow* window);       
+
+    private:
+        InputHandler();
+        static InputHandler input_handler_instance;
+        static GLFWwindow *_window;
+        static std::unordered_map<int, std::function<void()>> binded_table;
```

Comment:
key_to_function_table? I think it would be nice if the name also conveyed what the `int` here means.